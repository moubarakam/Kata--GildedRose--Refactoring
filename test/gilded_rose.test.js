const {
  Shop, 
  Item, 
  SulfurasItem,
  BackstageItem,
  ElixirItem,
  AgedBrieItem,
  DexterityVestItem
} = require("../src/gilded_rose");

const items = [
  new DexterityVestItem(10, 20),
  new AgedBrieItem(2, 0),
  new AgedBrieItem(2, 49),
  new ElixirItem(5, 7),
  new SulfurasItem(),
  new SulfurasItem(),
  new BackstageItem(15, 20),
  new BackstageItem(10, 49),
  new BackstageItem(5, 49),

  // This Conjured item does not work properly yet
  new Item("Conjured Mana Cake", 3, 6),
];

const days = Number(process.argv[7]) || 7;
const gildedRose = new Shop(items);
const shopBehaviour = [];

const cloneBehavior = JSON.parse(JSON.stringify(gildedRose.items));
describe("Gilded Rose Refactoring ###################################################", function() {
  it("We consider that the behavior of the initial code is the truth", function() {
    
    for (let day = 0; day < days; day++) {
      shopBehaviour.push(cloneBehavior);
      gildedRose.updateItem();
    }

    expect(shopBehaviour).toMatchSnapshot();
  });

  it("sellIn should be 999999", function() {
    // Given
    const gildedRose = new Shop([new SulfurasItem()]);

    // When
    const itemSulfura = gildedRose.updateItem();

    // Then
    expect(itemSulfura[0].sellIn).toBe(999999);
  });

});