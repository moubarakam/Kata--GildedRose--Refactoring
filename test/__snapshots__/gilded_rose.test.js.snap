// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose Refactoring ################################################### We consider that the behavior of the initial code is the truth 1`] = `
Array [
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
  Array [
    Object {
      "name": "+5 Dexterity Vest",
      "quality": 20,
      "sellIn": 10,
    },
    Object {
      "name": "Aged Brie",
      "quality": 0,
      "sellIn": 2,
    },
    Object {
      "name": "Aged Brie",
      "quality": 49,
      "sellIn": 2,
    },
    Object {
      "name": "Elixir of the Mongoose",
      "quality": 7,
      "sellIn": 5,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 999999,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 20,
      "sellIn": 15,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 10,
    },
    Object {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 49,
      "sellIn": 5,
    },
    Object {
      "name": "Conjured Mana Cake",
      "quality": 6,
      "sellIn": 3,
    },
  ],
]
`;
